packages/core/src/scp-cf/xsuaa-service.ts

```diff
@@ -145,22 +147,25 @@ export function fetchVerificationKeys(
 
 export function fetchVerificationKeys(
   xsuaaUriOrCredentials: string | XsuaaServiceCredentials,
-  clientId?: string,
+  clientIdOrJku?: string,
   clientSecret?: string
 ): Promise<TokenKey[]> {
+  // The case where the XsuaaServiceCredentials are given as object
   if (typeof xsuaaUriOrCredentials !== 'string') {
     return fetchVerificationKeys(
-      xsuaaUriOrCredentials.url,
+      clientIdOrJku || xsuaaUriOrCredentials.url,
       xsuaaUriOrCredentials.clientid,
       xsuaaUriOrCredentials.clientsecret
     );
   }
-
-  const url = `${xsuaaUriOrCredentials}/token_keys`;
```

Comment:
Just curious, have you tested whether the `/token_keys` is not needed anymore?

Reply:
no, but I thought that on the privider account the jku value and the xsuaa.url are the same and for the provider I tested it and got the JWT verified.

Reply:
But I will add a warn statement when the jku is not there because this should never be the case. I made the arguement optional to not create a breaking change.